### ISSUE TITLE
MacOS, 0.14 gives errors when changing the window size

### ISSUE THREAD

**Issue by pbgc** [alacritty/alacritty#8275]
### System

OS: macOS Ventura 13.7.1
Version: alacritty 0.14.0 (22a4475)

Launching alacritty and resizing the window or maximizing it gives me:

(running with RUST_BACKTRACE=full)

Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
~ ❯ thread 'main' panicked at library/std/src/io/stdio.rs:1118:9:
failed printing to stdout: Broken pipe (os error 32)
stack backtrace:
   0:        0x10534d910 - <std::sys::backtrace::BacktraceLock::print::DisplayBacktrace as core::fmt::Display>::fmt::h14dcb4794e811c9f
   1:        0x10535ec7c - core::fmt::write::h49a9cf68bfb4a321
   2:        0x1053518b4 - std::io::Write::write_fmt::h00e4f1b82d71ad85
   3:        0x105337f60 - std::panicking::default_hook::{{closure}}::hda52b8046522f561
   4:        0x105337b9c - std::panicking::default_hook::h9566cb73367aef3e
   5:        0x105338918 - std::panicking::rust_panic_with_hook::hfa41181adbcf532d
   6:        0x10534e218 - std::panicking::begin_panic_handler::{{closure}}::ha417e66bf9de7acf
   7:        0x10534db38 - std::sys::backtrace::__rust_end_short_backtrace::hc7b890f013fa9efb
   8:        0x105338054 - _rust_begin_unwind
   9:        0x10537e158 - core::panicking::panic_fmt::h29609029db4d41f4
  10:        0x105337544 - std::io::stdio::_print::he8373a8b6340dd41
  11:        0x10504b814 - fnm::commands::command::Command::call::h77387845411b7f95
  12:        0x105026198 - fnm::main::h24318726a9c2d8fb
  13:        0x105016b80 - std::sys::backtrace::__rust_begin_short_backtrace::h152c9580f53aceaf
  14:        0x105016b5c - std::rt::lang_start::{{closure}}::hef2de3ca26070566
  15:        0x10532e0a0 - std::rt::lang_start_internal::h31b5cf8a92666908
  16:        0x105027a04 - _main
~ ❯ thread 'thread 'rustc' panicked at library/std/src/io/stdio.rs:1118:9:
failed printing to stdout: Broken pipe (os error 32)
~ ❯ thread 'rustc' panicked at library/std/src/io/stdio.rs:1118:9:
failed printing to stdout: Broken pipe (os error 32)

**Comment by kchibisov** [comment]
I think it's _fish_ that made a panic, not the alacritty(otherwise it'd crash and close), so report to it to fish.

**Comment by pbgc** [comment]
@kchibisov ok. I will do it.... strange that this don't happen with all the other terminals I tried and don't happen if fish is running inside tmux (and alacritty)

**Comment by pbgc** [comment]
@kchibisov just another thing ... It's not strange that fish 3.7.1 - a C program .. is giving panics like: "thread 'main' panicked at library/std/src/io/stdio.rs:1118:9:" ???

Fish is being rewritten in rust for V4 .. but it was not yet launched ...

**Comment by kchibisov** [comment]
maybe something else that you do in the chain, like prompt if you use starship? We can not panic inside the terminal like that.

**Comment by pbgc** [comment]
@kchibisov just saw now that I had the problem with 0.13 also.
I don't use starship .. used it in the past but changed it to tide (pure fish).
I understand what you say (We can not panic inside the terminal like that.) ... and removed all by fish config and the problem went away ... BUT when I try the exact same config with macOS Terminal, iTerm2, Kitty and WezTerm and with none of them this problem happens ... If I had to bet ... I would bet that the cause is some problem with alacritty.
For now I changed to kitty.
Thanks for your answers!

**Comment by kchibisov** [comment]
that's clearly not alacritty problem though if something can not handle some event and given that removing things in your config _fixes_ it, it's clearly not our problem.

**Comment by pbgc** [comment]
@kchibisov what is giving problems is: https://github.com/IlanCosman/tide
Its something like starship but EVERYTHING is written in fish!

<img width="1316" alt="Captura de ecrã 2024-10-29, às 10 22 53" src="https://github.com/user-attachments/assets/5ff94490-44b0-4b13-b31a-46a683af4eff">

There is no way that this fish shell code is giving rust panics because there is no rust involved! but If I remove this fish functions ... the problem goes away!

**Comment by kchibisov** [comment]
maybe they call to something written in rust and there's some timing issue?

**Comment by pbgc** [comment]
@kchibisov I think it must be related to fnm (https://github.com/Schniz/fnm)
![Captura de ecrã 2024-10-29, às 18 57 51](https://github.com/user-attachments/assets/549791d1-d65c-4e63-b5a6-e53dc90f35c5)

Maybe tide checking if node is present in the current folder (or a project) to display the version etc.
Like I told before .. this problem does not occur if I use tmux: fish -> tmux -> alacritty and does not occur in any other terminal I tried (macOS terminal, iTerm2, Kitty, WezTerm).

I understand this is not really a priority and you don't even consider this an alacritty problem! I'm just giving some more information ... and like I said ... as I don't want to change my workflow ... I changed to kitty.
